### PR TITLE
README: removed todo, added section about strings and missing features

### PR DESCRIPTION
**Pull request by giann**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,94 +7,40 @@
...
 
-### `dv = lua_todataview(L, idx)`
+Fengari implements Lua 5.3 semantics and will hopefully follow future Lua releases. If you find any noticeable difference between Fengari and Lua's behaviours, please [report it](https://github.com/fengari-lua/fengari/issues).
 
-Equivalent to `lua_tolstring` but returns a [`DataView`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) instead of a string.
+### Strings
+
+Lua strings are 8-bits clean and can embed `\0`. Which means that invalid UTF-8/16 strings are valid Lua strings. Lua functions like `string.dump` even use strings as a way of storing binary data.
 
+To address that issue, Lua strings are represented by an array of bytes in Fengari. To push a JS string on the stack you can use `lua_pushliteral`. To get a Lua string on the stack as a JS string you can use `lua_tojsstring`.
```
Comment (daurnimator) [inline README.md]: Add note here about utf16 conversion

File: README.md [hunk 2]
```diff
@@ -7,94 +7,40 @@
...
+    - The entire `io` lib
+    - `os.remove`
+    - `os.rename`
+    - `os.tmpname`
+    - `os.execute`
 
-### `lua_tojsstring(L, idx)`
 
-Alias for `lua_toljsstring`.
+### `dv = lua_todataview(L, idx)`
```
Comment (daurnimator) [inline README.md]: I think put these extra JS functions under different heading

File: README.md [hunk 3]
```diff
@@ -7,94 +7,41 @@
...
 
-Equivalent to `lua_tolstring` but returns a [`DataView`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) instead of a string.
+To address that issue, Lua strings are represented by an array of bytes in Fengari. To push a JS string on the stack you can use `lua_pushliteral` which will convert it to an array of bytes before pushing it. To get a Lua string on the stack as a JS string you can use `lua_tojsstring` which will attempt to convert it to a UTF-16 JS string. The latter won't give you what you expect if the Lua string is not a valid UTF-16 sequence.
 
 
-### `str = lua_toljsstring(L, idx)`
+### _Missing_ features
 
-Equivalent to `lua_tolstring` but returns the string as a JavaScript string (as if `to_jsstring()` was called on the result).
+- `lua_gc/collectgarbage`: Fengari relies on the JS garbage collector and does not implement its own.
```
Comment (daurnimator) [inline README.md]: Have `lua_gc` and `collectgarbage` these as two separate markdown code things

Comment (giann) [inline README.md]: done

File: README.md [hunk 4]
```diff
@@ -7,94 +7,41 @@
...
+Fengari implements Lua 5.3 semantics and will hopefully follow future Lua releases. If you find any noticeable difference between Fengari and Lua's behaviours, please [report it](https://github.com/fengari-lua/fengari/issues).
 
-## Extensions
+### Strings
 
-### `dv = lua_todataview(L, idx)`
+Lua strings are 8-bits clean and can embed `\0`. Which means that invalid UTF-8/16 strings are valid Lua strings. Lua functions like `string.dump` even use strings as a way of storing binary data.
 
-Equivalent to `lua_tolstring` but returns a [`DataView`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) instead of a string.
+To address that issue, Lua strings are represented by an array of bytes in Fengari. To push a JS string on the stack you can use `lua_pushliteral` which will convert it to an array of bytes before pushing it. To get a Lua string on the stack as a JS string you can use `lua_tojsstring` which will attempt to convert it to a UTF-16 JS string. The latter won't give you what you expect if the Lua string is not a valid UTF-16 sequence.
```
Comment (daurnimator) [inline README.md]: Mention `lua.to_string` and `lua.to_jsstring`?

Comment (daurnimator) [inline README.md]: Also perhaps we should only expose these under `fengari`? not `fengari.lua`?

Comment (giann) [inline README.md]: Wasn't even sure anymore if they were at all exposed to the user

File: README.md [hunk 5]
```diff
@@ -7,94 +7,41 @@
...
 
 
-### `str = lua_toljsstring(L, idx)`
+### _Missing_ features
 
-Equivalent to `lua_tolstring` but returns the string as a JavaScript string (as if `to_jsstring()` was called on the result).
+- `lua_gc`/`collectgarbage`: Fengari relies on the JS garbage collector and does not implement its own.
+- The following functions are only available in Node:
```
Comment (daurnimator) [inline README.md]: weak tables are currently not supported.

**Pull request merged**